### DISCUSSION TITLE
hping has been disabled

### DISCUSSION THREAD

**Discussion by DLewis01** [Homebrew/discussions#5555] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: fa53e7b1e51a2deb7ec5a1e12452a1182dc342f7
Last commit: 25 hours ago
Core tap JSON: 13 Aug 08:28 UTC
Core cask tap JSON: 13 Aug 08:28 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.45.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-platformdirs
  python-distlib
  python-filelock

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  python-packaging
  python@3.8
  pyyaml

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python-packaging
  cryptography
  numpy
  python-platformdirs
  python-filelock
```

### Description of issue

When trying to install hping, I get:

Error: hping has been disabled because it is not maintained upstream! It will be disabled on 2024-02-15.

Can this be reconsidered. Just because it isnt being developed further does not mean there isnt a use for it, nor that it is no longer used. It has all the features it's needs, and just works.

Failing this, I'll have to consider moving back to macports which still has this in it's repo.

**Comment by SMillerDev** [comment]
> Can this be reconsidered. Just because it isnt being developed further does not mean there isnt a use for it, nor that it is no longer used. It has all the features it's needs, and just works.

There might be a use for it, but it might also have security issues and become increasingly difficult to maintain. That's why Homebrew won't ship it in the default tap anymore. But you can consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by daeho-ro** [comment]
The homepage and source url is also down, so even if it is not disabled, I think the formula cannot be installed.

**Reply by DLewis01** [reply]
a quick google show source is still available in multiple locations 
https://github.com/antirez/hping?tab=readme-ov-file

**Reply by SMillerDev** [reply]
If you can find which is official and make a pull request to add it I'll gladly review it.

**Reply by daeho-ro** [reply, marked as answer]
@DLewis01 Can you try
```
brew install GreedyLabs/tap/hping
```
and please check the functionality.

**Reply by DLewis01** [reply]
Thank you, it is working perfectly.

**Reply by p-linnane** [reply]
Please note that is not an official tap, so any issues you encounter will need to be discussed with the maintainer of the tap, rather than Homebrew's maintainers.

**Comment by parsa222** [comment]
`brew install draftbrew/tap/hping` doesn't work anymore.

The DraftBrew org (https://github.com/draftbrew) returns a 404 not found as of 2026-07-28.

I made a replacement, which also supports the Tcl scripting (forked from the same source: https://github.com/antirez/hping):

```sh
brew install parsa222/tap/hping3
```

https://github.com/parsa222/homebrew-tap/blob/master/README.md
